x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/nio/SSLChannelContextTests.java

```diff
@@ -427,4 +428,32 @@ public int consumeReads(InboundChannelBuffer channelBuffer) throws IOException {
             return fn.apply(channelBuffer);
         }
     }
+
+    private static class TestSelectionKey extends AbstractSelectionKey {
```

Comment:
NIT: Might as well just use `mock(AbstractSelectionKey.class)` instead and save a few lines here?